clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/provider/agent/AmazonApplicationLoadBalancerCachingAgent.groovy

```diff
@@ -95,25 +132,103 @@ class AmazonApplicationLoadBalancerCachingAgent extends AbstractAmazonLoadBalanc
+        return null
+      }
+    }
+    if (loadBalancer == null) { loadBalancer = [:] }
+
+    List<TargetGroup> targetGroups = metricsSupport.readData {
+      try {
+        return loadBalancing.describeTargetGroups(
+          new DescribeTargetGroupsRequest().withLoadBalancerArn(loadBalancer?.loadBalancerArn)
+        ).targetGroups
+      } catch (LoadBalancerNotFoundException ignored) {
+        return []
+      } catch (TargetGroupNotFoundException ignored) {
         return []
       }
     }
 
-    def cacheResult = metricsSupport.transformData { buildCacheResult(loadBalancers, [:], System.currentTimeMillis(), []) }
+    // Get all the target group health and attributes
+    Map<String, List<TargetHealthDescription>> targetGroupArnToHealths = metricsSupport.readData {
+      targetGroups.collectEntries { targetGroup ->
+        List<TargetHealthDescription> targetHealthDescriptions = loadBalancing.describeTargetHealth(
+          new DescribeTargetHealthRequest().withTargetGroupArn(targetGroup.targetGroupArn)
+        ).targetHealthDescriptions
+        return [(targetGroup.targetGroupArn): targetHealthDescriptions]
+      }
+    }
+
+    Map<String, List<TargetGroupAttribute>> targetGroupArnToAttributes = metricsSupport.readData {
+      targetGroups.collectEntries { targetGroup ->
+        List<TargetGroupAttribute> targetGroupAttributes = loadBalancing.describeTargetGroupAttributes(
+          new DescribeTargetGroupAttributesRequest().withTargetGroupArn(targetGroup.targetGroupArn)
+        ).attributes
+        return [(targetGroup.targetGroupArn): targetGroupAttributes]
+      }
+    }
+
+    Map<String, List<Listener>> loadBalancerArnToListeners = [:]
+    Map<Listener, List<Rule>> listenerToRules = new HashMap<>()
+
+    // Add the listeners
+    List<Listener> listenerData = new ArrayList<>()
+    try {
+      DescribeListenersRequest describeListenersRequest =
+        new DescribeListenersRequest(loadBalancerArn: loadBalancer.loadBalancerArn)
+      while (true) {
+        DescribeListenersResult result = loadBalancing.describeListeners(describeListenersRequest)
+        listenerData.addAll(result.listeners)
+        if (result.nextMarker) {
+          describeListenersRequest.withMarker(result.nextMarker)
+        } else {
+          break
+        }
+      }
+    } catch (LoadBalancerNotFoundException ignore) {
+      // this is acceptable since we may be waiting for the caches to catch up
+    }
+    loadBalancerArnToListeners.put(loadBalancer.loadBalancerArn, listenerData)
+
+    for (listener in listenerData) {
+      listenerToRules[listener] = []
+      try {
+        DescribeRulesRequest describeRulesRequest = new DescribeRulesRequest(listenerArn: listener.listenerArn)
+        DescribeRulesResult result = loadBalancing.describeRules(describeRulesRequest)
+        listenerToRules.get(listener).addAll(result.rules)
+      } catch (ListenerNotFoundException ignore) {
+        // should be fine
```

Comment:
should be fine eh?

I'm somewhat concerned about blindly catching exceptions and not even logging ... it makes debugging more of a challenge.

I'd be more inclined to fail if something we expect to exist doesn't ... ?